### PR TITLE
OneOf support

### PR DESCRIPTION
**Pull request by gzigzigzeo**
Enables OneOf support. Fixes bug with messages having no fields.

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]
I tested this branch against Teleport API V10 and it generated the schema correctly

**Comment by marcoandredinis** [comment]
@r0mant Can you please take a look?

**Comment by marcoandredinis** [comment]
@r0mant ping
(sorry, we really need this to upgrade the API version we use in Teleport Plugins)

**Review by r0mant — APPROVED** [review]
Sorry about the delay.

**Comment by marcoandredinis** [comment]
We found an issue with `oneOf` when one of the branches has 0 attributes
@gzigzigzeo is aware of it and will work on it

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.0.0
\ No newline at end of file
+v1.2.0
```
Comment (marcoandredinis) [inline VERSION:1]: :nail_care: 
```suggestion
v1.1.0
```

**Comment by marcoandredinis** [comment]
Just tested and it worked as expected :+1:

**Pull request merged**